### DISCUSSION TITLE
Calculating the wind direction as each turbine.

### DISCUSSION THREAD

**Discussion by JackDI1** [NatLabRockies/floris#728] [Q&A] [answered]
Hi,

Is it possible to calculate the wind direction at a given set of coordinates? I am currently using sample_flow_at_points to get the wind speed at each turbine, but is there a similar method to calculate the wind direction at each turbine?

Thanks,
Jack

**Comment by rafmudaf** [comment]
Hi @JackDI1 can you describe how you input the wind direction through your input file or script?

**Reply by JackDI1** [reply]
Hi @rafmudaf sure, the wind direction is just a single value set in an input file. 

```
  ###
  # The wind directions to include in the simulation.
  # 0 is north and 270 is west.
  wind_directions:
  - 30.0
```

**Reply by rafmudaf** [reply]
I asked because there's an option to specify wind directions at various locations and let FLORIS interpolate between the given locations. However, in your case the wind direction is uniform throughout the wind field. So, the short answer is no there's not a method to probe the flow field for the wind direction, but you should know the wind direction because it is a user-given input.

Does this answer your question or have I misunderstood the issue?

**Reply by JackDI1** [reply]
Hi @rafmudaf, sorry for the delay getting back to you!

I was after the wind direction at each turbine following a change in yaw angle offset. This is a calculation that can be done simply in code (that I was overcomplicating in my mind!) but I was wondering if it was already being done in floris. 

Just to confirm the yaw angles in Floris are the nacelle direction relative to the wind direction, e.g. nacelle - wind direction?

Thanks,
Jack

**Reply by rafmudaf** [reply, marked as answer]
@JackDI1 The wind direction in FLORIS is set by you and it does not change after you've changed the yaw angle. If you are referring to the induction zone, this is not currently modeled in FLORIS.

> Just to confirm the yaw angles in Floris are the nacelle direction relative to the wind direction, e.g. nacelle - wind direction?

Yes, that's correct. Currently, FLORIS only supports a single user-given wind direction that is constant for the entire flow field.

**Reply by JackDI1** [reply]
Ok thank you, that is helpful.

**Reply by rafmudaf** [reply]
Great, thanks for following up and selecting an answer. Let me know if I've still missed some part of your question.
